js/src/forum/components/addManagePoliciesOption.js

```diff
@@ -0,0 +1,47 @@
+import { extend } from 'flarum/common/extend';
+import app from 'flarum/forum/app';
+import SettingsPage from 'flarum/forum/components/SettingsPage';
+import FieldSet from 'flarum/common/components/FieldSet';
+import Switch from 'flarum/common/components/Switch';
+
+async function updatePolicy(policy, value) {
+  const endpoint = `${app.forum.attribute('apiUrl')}${policy.apiEndpoint()}${value ? '/accept' : '/decline'}`;
+  const updated = await app.request({
+    url: endpoint,
+    method: 'POST',
+  });
+  app.store.pushPayload(updated);
+}
+
+export default function () {
+  extend(SettingsPage.prototype, 'settingsItems', function (items) {
+    const policies = app.store.all('fof-terms-policies').filter((policy) => policy.optional());
+
+    let policyState = app.session.user.fofTermsPoliciesState();
+
+    items.add(
+      'policies',
+      <FieldSet label={'Policies'}>
+        {policies.map((policy) => {
+          const { is_accepted } = policyState[policy.id()];
+          return (
+            <div class={'Fof-Terms-Policy-User-Settings-Management'}>
+              <Switch
+                state={is_accepted}
+                onchange={async (value) => {
+                  policyState[policy.id()].is_accepted = value;
+                  this[policy.form_key()] = value;
+                  await updatePolicy(policy, value);
+                  m.redraw();
+                }}
+              >
+                <a href={policy.url() || ''}>{policy.name()}</a>
```

Comment:
If a URL is provided, this should be opened in a new tab with `target="_blank"`. If no URL is provided by the admin, I think it should be just a regular text. But I'm curious what others are thinking about this...

Reply:
Proposed changes were committed.